src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -260,6 +281,65 @@ private async Task<Document> InlineTemporaryAsync(Document document, VariableDec
             return updatedDocument;
         }
 
+        private static bool HasPossibleSideEffects(SyntaxNode expression, SyntaxNode syntaxRoot, IEnumerable<ReferenceLocation> references)
```

Comment:
why did you write your own helper for ths?  i thought we had one already.

Reply:
specifically: UseCompoundAssignmentUtilities.IsSideEffectFree.  Can we ot move that to another location and share between both of these?

Reply:
I was looking into that helper, but it doesn't seem like it really checks the criteria that we want and instead seems specifically designed for compound assignments. I even tried using it in the code at first, but pretty much everything was deemed as non-side effect free, even simple literal expressions like "1".

Reply:
>  but pretty much everything was deemed as non-side effect free, even simple literal expressions like "1".

Well, we can always improve on that helper :D

Reply:
Would it be better to expand on the helper despite that the two refactoring scenarios seem to differ in the necessary checks and scenarios?

For one thing, UseCompoundAssignmentUtilities.IsSideEffectFree seems to apply to the scenario where the original expression is changed from being executed twice to once. Inline temporary variable seems to go in the opposite direction, where the original expression is changed from being executed once to being executed multiple times.

Additionally, for inline temporary variable, I don't know if it's necessary to care about all the cases that UseCompoundAssignmentUtilities.IsSideEffectFree checks for. We just need to know whether we're dealing with an object creation expression or invocation expression and whether a property/method is accessed.

UseCompoundAssignmentUtilities.IsSideEffectFree checks for things like whether there's multiple reads of a property, which doesn't seem like it would be necessary in for the inline temporary variable refactoring.